index.html

```diff
@@ -745,10 +752,9 @@ <h3>Validate Algorithm</h3>
 |credentialIndex| times the |size| in the |bitstring|.
         </li>
         <li>
-For `statusPurpose` of `revocation` or `suspension`,
-return `true` if `status` is `1`, and return `false`
-if `status` has any other value.  For other `statusPurpose`,
-return the corresponding `value` of the `status`
+For `statusPurpose` of `revocation` or `suspension`, return `true` if |status|
+is `1`, and return `false` if |status| has any other value. If the
+`statusPurpose` is `message`, return the corresponding `message` of the `value`
 as indicated in the `statusMessages` array.
```

Comment:
I'm not sure we want to call this a "Validate algorithm" given other discussions and the conflation with business rules, but setting that to the side, I think it would better if we made this algorithm more consistent in what it returns.

We could have it return an object (`map`) with "valid" being a boolean as well as the current status value and, optionally, a `message`. I think this would be more consistent and more easily consumable by simpler software -- and other extensions could add their own properties. I also think the current text on `revocation` and `suspension` is reversed a bit from what we'd want. So perhaps something more like this:

```suggestion
        </li>
Let |result| be an empty [=map=].
        <li>
Set the `status` key in |result| to |status|, and set the `purpose` key in
|result| to the value of `statusPurpose`.
        </li>
        <li>
If |status| is `0` set the `valid` key in |result| to `false`, otherwise set it to
`true`.
        </li>
        <li>
If the `statusPurpose` is `message`, set the `message` key in |result| to
the corresponding `message` of the `value` as indicated in the `statusMessages` array.
        </li>
        <li>
Return |result|.
```

Reply:
I think this is an improvement. Tweak punctuation (change comma to semicolon; add 2 other commas) in line 762 to --
`If |status| is `0`, set the `valid` key in |result| to `true`; otherwise, set it to`

Reply:
Done.